### PR TITLE
feat: installing tailwind and setting up folder structure

### PR DESCRIPTION
**Pull request by mcturco**
**Description**
This PR introduces the following changes:
- Use postcss to configure tailwindcss with storybook
- Create folder structure for all 3 design systems: Documentation, Tooling, Website
- Created Introduction MDX files for each repo (will need to add more copy here in the future)
- Added our first component in `Website`, the `Heading` component with all of its states/stories. We will need to still document this further, this is just a boilerplate for now

Screen Shot 2022-03-21 at 11 08 45 AM

cc: @yashsehgal, I was exploring the Storybook docs and thought that this was a good way to organize the components, but of course if you have any other ideas let me know and we can discuss them!

-----

**And if you are interested, I have recorded an explanation of these changes:**

image
[@asyncapi/design-system: PR #17 - Watch Video](https://www.loom.com/share/5461e4f5f9204f80bd8ccad9c4bab4d0)

### CODE REVIEW

File: src/website/typography/Heading/Heading.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+export default function Heading({
+    typeStyle = "heading-lg",
```
Comment (fmvilas) [inline src/website/typography/Heading/Heading.js:2]: I'd remove the word `heading` from all the `typeStyle` options. It's a bit repetitive and unnecessary since we already know it's for heading because we're on the heading component :)

File: src/website/typography/Heading/Heading.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        case 'body-lg':
+            classNames = `font-heading text-lg leading-relaxed tracking-body font-regular ${className || ''}`
+            break;
+        case 'body-md':
+            classNames = `font-heading text-base leading-relaxed tracking-body font-regular ${className || ''}`
+            break;
+        case 'body-sm':
+            classNames = `font-heading text-sm leading-relaxed tracking-body font-regular ${className || ''}`
+            break;
+        default:
```
Comment (fmvilas) [inline src/website/typography/Heading/Heading.js:44]: When is the default option coming into play?

File: src/website/typography/Heading/Heading.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+export default function Heading({
+    typeStyle = "heading-lg",
+    level = 'h2',
+    textColor = 'text-primary-800',
```
Comment (fmvilas) [inline src/website/typography/Heading/Heading.js:4]: Maybe `textColor` should be decomposed into two props? Something like `color` (primary) and `shade` (800)? We'll want to avoid people passing random class names that are not related to text color here.

File: src/website/typography/Heading/Heading.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    className,
+    children
+}) {
+    let classNames = ''
+    const Tag = `${level}`;
```
Comment (fmvilas) [inline src/website/typography/Heading/Heading.js:9]:
```suggestion
    const Tag = level;
```

But we should check we only get a valid value here and not some random string.

**Review by fmvilas — CHANGES_REQUESTED** [review]
This is looking great! Left a few suggestions.

**Comment by mcturco** [comment]
> This is looking great! Left a few suggestions.

Hey @fmvilas! I'm not sure about making component structure changes within here just yet... I actually did want to talk with you and @magicmatatjahu about how we can connect this repo to components in other repos so when we make changes here that it will automatically release those updates to the repos the components are used in. 

Maybe for now if we want to make these changes to the component we should do this within the [@asyncapi/website](https://github.com/asyncapi/website) repo? These components were merged with my new styles PR. I do agree with your feedback but not sure how we should implement those changes. What do you think?

**Comment by mcturco** [comment]
Cool! I can also create a separate issue for that!

**Comment by mcturco** [comment]
Issue created here #18

**Comment by magicmatatjahu** [comment]
There are no problems with the distribution of components in this repo. Our design system is a normal package that we will publish to npm. So the whole flow will be like this:

- website uses a new package - let's call it `@asyncapi/design-system` for the moment
- someone contributes to this repository with a new component, e.g. `Tabs` component
- After merging, the whole release process starts and the new version of `@asyncapi/design-system` with the new component is published
- people can now use the `Tabs` component in the website repo

if there are any inaccuracies, please let me know :)

**Comment by magicmatatjahu** [comment]
Additionally it should be possible to connect this repository with Netlify to build storybook on PR and to check alive changes :)

**Comment by magicmatatjahu** [comment]
Looking at the code and Fran's comments, I have nothing additional to add for now. The only thing is that we can switch to TypeScript as soon as possible 😄

**Comment by mcturco** [comment]
@magicmatatjahu 

> So the whole flow will be like this:

The flow you mentioned here is exactly what I had in mind! 😄  I am actually wondering too if we should create 3 separate packages, 1 package for each sub-system of the AsyncAPI design system. So maybe it would look like this:

- Website Components `@asyncapi/designsystem-website`
- Documentation Components (Generator) `@asyncapi/designsystem-docs`
- Tooling Components (Tooling UI) `@asyncapi/designsystem-tooling`

Eventually `designsystem` will change to a single-word name to represent the AsyncAPI design system. For example, you had suggested the name: `Elon` 😆  So maybe the website components package would be named: `@asyncapi/elon-website` Just suggestions!

> Additionally it should be possible to connect this repository with Netlify to build Storybook on PR and to check alive changes :)

Agreed! I don't know how to do that, so maybe you can help with it?

> Looking at the code and Fran's comments, I have nothing additional to add for now. The only thing is that we can switch to TypeScript as soon as possible 😄

Are you talking about the component that I added here so far or just all of the components in general? I will definitely step aside for that process as I am not the most familiar with React JS/TypeScript just yet, but I will still definitely work with you on this!

**Comment by mcturco** [comment]
@magicmatatjahu @fmvilas 

So for next steps of merging this PR, I want to point out that the purpose really for this PR was to add tailwindcss support to storybook. The other part where I added some component/story files doesn't matter too much as it will all change in the near future when we are ready to release the design system as a whole (with current components, not yet new components, that will take a longer amount of time) That way when we want to iterate on these components, it would be easy to do so from this repo and then release the update with the documentation of updates/breaking changes.

So if you all accept the changes as far as tailwindcss/postcss config, I think we can approve and merge this PR and then work on the next steps for this repo 😄

**Comment by fmvilas** [comment]
So if I understand correctly, @mcturco what you want to do is to "only" design (with code) the components here and those will eventually be reworked/revisited to improve coding style/practices?

**Comment by mcturco** [comment]
> what you want to do is to "only" design (with code) the components here

@fmvilas I'm not sure if by this you mean only design here and not in figma, but if that's what you meant, that's not what I am wanting to do. I think we still need design mock ups in figma, and then when we are ready to build and test them they would get coded and added into this repo. It is also important that we keep components in this repo in sync always with the Design System components library in Figma.

So I guess we need to edit the flow suggested by @magicmatatjahu to this:

- website uses a new package - let's call it `@asyncapi/design-system` for the moment
- design process begins for new `Tabs` component: Figma -> Build (this repo) -> Test and iterate through that cycle until we are ready to release (perhaps releases will include several new components at once)
- New `Tabs` component is documented within storybook
- After merging, the whole release process starts and the new version of `@asyncapi/design-system` with the new component is published
- people can now use the `Tabs` component in the website repo
someone contributes to this repository with a new component, e.g. Tabs component

Of course, just using `Tabs` as an example here 😄  And this is just what I think might work as far as a design -> release process goes, but feel free to leave your suggestions because I am still learning what the best workflow for design systems looks like!
